mmv1/products/storageinsights/DatasetConfig.yaml

```
@@ -0,0 +1,294 @@
+# Copyright 2025 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: 'DatasetConfig'
+description: |
+    Represents a Storage Insights DatasetConfig.
+references:
+  guides:
+    'Official Documentation': 'https://cloud.google.com/storage/docs/insights/datasets'
+  api: 'https://cloud.google.com/storage/docs/insights/reference/rest/v1/projects.locations.datasetConfigs'
+
+
+base_url: 'projects/{{project}}/locations/{{location}}/datasetConfigs'
+self_link: 'projects/{{project}}/locations/{{location}}/datasetConfigs/{{dataset_config_id}}'
+create_url: 'projects/{{project}}/locations/{{location}}/datasetConfigs?datasetConfigId={{dataset_config_id}}'
+
+custom_code:
+  post_create: templates/terraform/post_create/storage_insights_dataset_config.go.tmpl
+  pre_update: templates/terraform/pre_update/storage_insights_dataset_config.go.tmpl
+
+update_verb: 'PATCH'
+
+# Constructing updateMask in pre_update due to API exceptions.
+update_mask: false
+
+# If true, code for handling long-running operations is generated along with
+# the resource. If false, that code is not generated.
+# write operations of dataset config is an LRO.
+autogen_async: true
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    base_url: '{{op_id}}'
+
+import_format:
+  - 'projects/{{project}}/locations/{{location}}/datasetConfigs/{{dataset_config_id}}'
+
+examples:
+  - name: 'storage_insights_dataset_config'
+    primary_resource_id: 'config'
+    vars:
+      dataset_config_id: 'my_config'
+
+virtual_fields:
+  - name: 'link_dataset'
+    type: Boolean
+    default_value: false
+    description: |
+      A boolean terraform only flag to link/unlink dataset.
+
+      -> **Note** This is a virtual terraform only field used to link/unlink a dataset config.
+      This field is not part of the actual terraform resource.<br>
```

```suggestion
      -> **Note** This is a virtual terraform only field used to link/unlink a dataset config.
      This field is not part of the actual terraform resource.<br>
```
I'd remove these two lines. They're not end-user focused